test/quantization/core/test_workflow_ops.py

```diff
@@ -293,17 +293,6 @@ def test_backward_per_tensor(self, device, X):
         Y_prime.backward(dout)
         np.testing.assert_allclose(dX.cpu(), X.grad.cpu().detach().numpy(), rtol=tolerance, atol=tolerance)
 
-    def test_forward_backward_per_tensor_with_amp(self):
-        net = nn.Sequential(nn.Conv2d(1, 1, 3))
-        net.qconfig = torch.ao.quantization.get_default_qat_qconfig('fbgemm')
-        net_prep = torch.ao.quantization.prepare_qat(net)
-
-        with torch.cuda.amp.autocast():
-            x = torch.randn(4, 1, 5, 5)
-            out = net_prep(x).sum()
-            out.backward()
-            self.assertTrue(net_prep[0].weight.grad is not None)
```

Comment:
cc @jerryzh168. I think  this quantization test is broken. It's activating cuda's AMP (which uses the `AutocastCUDA` dispatch key), but running the test with CPU tensors, which means that AMP will never actually be triggered.

With this change, activating any of the autocast API's (either cuda or CPU) will trigger autocast for **all** tensors, using whatever autocast dtype is chosen for the particular backend.

Is this test reasonable to delete? I tried switching the tensors to be cuda, and also switching the autocast API to use the cpu version, and in both cases the test fails (a quantized op was expecting a different dtype). So I just removed the test.

Reply:
@ngimel also just wanted to double check with you. That's probably technically BC-breaking (changing what happens when you try to use CPU tensors under a CUDA autocast context manager). Do you think that's behavior users might be relying on? Or is that just mis-using the API's in a way that's acceptable for us to change.

If we can't change it, we might just have to live with each backend needing its own autocast key.

Reply:
I don't have much context on this test, cc @supriyar @vkuzo do you know who added this test?

Reply:
The test was added by https://github.com/pytorch/pytorch/pull/52612. Looks like the intent of the test case was to check if `torch.fake_quantize_per_tensor` worked correctly with AMP, and it did not realize AMP would not be called without CUDA tensors.  I think it's fine to either delete this test or change it to use CUDA tensors.